source/extensions/filters/network/mysql_proxy/mysql_codec_clogin.cc

```diff
@@ -238,6 +286,17 @@ void ClientLogin::encodeResponse41(Buffer::Instance& out) const {
     BufferHelper::addString(out, auth_plugin_name_);
     BufferHelper::addUint8(out, enc_end_string);
   }
+  if (client_cap_ & CLIENT_CONNECT_ATTRS) {
+    Buffer::OwnedImpl conn_attr;
+    for (const auto& kv : conn_attr_) {
+      BufferHelper::addLengthEncodedInteger(conn_attr, kv.first.length());
+      BufferHelper::addString(conn_attr, kv.first);
+      BufferHelper::addLengthEncodedInteger(conn_attr, kv.second.length());
+      BufferHelper::addString(conn_attr, kv.second);
+    }
+    BufferHelper::addLengthEncodedInteger(out, conn_attr.length());
+    out.move(conn_attr);
```

Comment:
What happens if `out` consists of some data added before?

Reply:
I do not consider it, I will create a buffer at begin of the function, and `out.move(buf)` at last.

Reply:
I'm not quite understand, `out.append(conn_attr.length());out.move(conn_attr);` just appends length and connection attributes to `out`.